pkg/sidecar-controller/snapshot_controller.go

```diff
@@ -63,13 +64,13 @@ func (ctrl *csiSnapshotSideCarController) syncContent(content *crdv1.VolumeSnaps
 			// underlying storage system. Note that the deletion snapshot operation will
 			// update content SnapshotHandle to nil upon a successful deletion. At this
 			// point, the finalizer on content should NOT be removed to avoid leaking.
-			return ctrl.deleteCSISnapshot(content)
+			return false, ctrl.deleteCSISnapshot(content)
```

Comment:
I think it's better to check the return code of ctrl.deleteCSISnapshot(content) and set requeue to true if err != nil. This will make the code easier to read.